File: backend/src/UnCRM.Api/Domain/Models/Usuario.cs
using UnCRM.Api.Domain.Enums;

namespace UnCRM.Api.Domain.Models
{
    public class Usuario : Entidade<long>
    {
        public string Login { get; set; } = string.Empty;
        public string Senha { get; set; } = string.Empty;
        public string Nome { get; set; } = string.Empty;
        public CargoEnum Cargo { get; set; }
        public DateTime DataCadastro { get; set; }
        public DateTime? DataInativacao { get; set; }
    }
}

File: frontend/uncrm/src/models/Usuario.js
export default class Usuario {
  constructor(dados = {}) {
    this.id = dados.id;
    this.login = dados.login;
    this.nome = dados.nome;
    this.senha = dados.senha;
    this.dataInativacao = dados.dataInativacao;
    this.dataCadastro = dados.dataCadastro;
    this.token = dados.token;
    this.cargo = dados.cargo;
  }

  static cargoMap = {
    0: "Gerente",
    1: "Supervisor",
    2: "Atendente",
  };

  get cargoDescricao() {
    return Usuario.cargoMap[this.cargo] || "Desconhecido";
  }
  
  get isGerente() {
    return this.cargo === 0;
  }

  get dataCadastroFormatada() {
    return this.dataCadastro
      ? new Date(this.dataCadastro).toLocaleDateString("pt-BR")
      : "-";
  }

  get statusItem() {
    return this.dataInativacao ? "Inativo" : "Ativo";
  }
}
